src/index.ts

```diff
@@ -289,6 +289,32 @@ function main() {
                     }
                     throw new Error('Locations list cannot be empty');
                 })
+                .then((ans: Answers) => {
+                    // Use Cosmos DB for telemetry storage for China environment
+                    if (program.environment === AzureEnvironment.AzureChina.name) {
+                        return Promise.resolve(ans);
+                    }
+                    // Check if the selected location support Time Series Insights resource type for Global environment
+                    // Use the default value of template when the location does not support it
+                    const resourceManagementClient = new ResourceManagementClient(
+                        new DeviceTokenCredentials(cachedAuthResponse.credentials),
+                        answers.subscriptionId,
+                        baseUri);
+
+                    ans.telemetryStorageType = 'tsi';
```

Comment:
Do we default to tsi now?

Reply:
yes, for global regions since we can use one default region now. And for Mooncake, we will use cosmos DB as default in the template.